### ISSUE TITLE
Tax Summary with multiple rates doesnt show details on multiple lines

### ISSUE THREAD

**Issue by marieevekroy** [classyllama/ClassyLlama_AvaTax#70]
**Steps to reproduce**
In Stores > Config > Sales > Tax, set "Display Full Tax Summary" to Yes
In cart/checkout, choose an address where multiple rates are applicable

**Actual result**
When you expand the summary, it shows all the rates with the percentage, with only the tax total on the first line.

**Expected result**
When you expand the summary, in cart, checkout, in order admin panel, it should show the detailed amount of tax $$ per rate ($1.00 on the first line, $1.99 on the second line) instead of $2.99

![capture d ecran 2017-07-04 a 10 15 27](https://user-images.githubusercontent.com/29894781/27833898-af9f2efa-60a2-11e7-8754-7e0d7b3f1af0.png)
![capture d ecran 2017-07-04 a 10 14 09](https://user-images.githubusercontent.com/29894781/27833896-af9dac24-60a2-11e7-8661-e1b777b4f663.png)

**Comment by jeremybouvet** [comment]
I'm having the same issue !

**Comment by erikhansen** [comment]
Unfortunately, Magento's native "tax summary" does _not_ support displaying the prices on each individual row. I'm the one who built this part of the extension and we're feeding data into Magento in the exact same way that native tax rates are being fed into the system.

As a point of comparison, I turned off AvaTax and configured some native tax rates to try and emulate the behavior to see how the native tax summary was displayed.

Screenshots of configuration:

![magento admin-kuuax](https://user-images.githubusercontent.com/129031/28023432-85e46502-6553-11e7-883d-1f4f9d99125d.png)


![magento admin-unlk6](https://user-images.githubusercontent.com/129031/28023449-8ff00bfa-6553-11e7-9796-d703839886e5.png)

![09-39-02 shopping cart-708jm](https://user-images.githubusercontent.com/129031/28023480-a2d05cca-6553-11e7-83de-eb3ab496a2d2.png)

Since this is not an issue with this extension, I'm closing this issue. Feel free to submit an issue or PR to the Magento 2 Github.

**Comment by marieevekroy** [comment]
Hi Erik,

Unfortunately, your settings are not quite correct for the example I try to explain. Native Magento works well.

In Tax Rules, when you have multiple rules applicable, the totals summary display correctly on two lines.

<img width="1108" alt="capture d ecran 2017-07-12 a 15 08 52" src="https://user-images.githubusercontent.com/29894781/28136409-9ffea0d6-6717-11e7-9cf4-beb481111bfc.png">
<img width="258" alt="capture d ecran 2017-07-12 a 15 09 15" src="https://user-images.githubusercontent.com/29894781/28136404-994cfd14-6717-11e7-8979-bf46d7deb6e9.png">

From what I understand, the problem lies somewhere in TaxCalculation class where the data received from Avatax is converted in Magento object. All taxes are converted in only one applied tax. 

With the same example of Quebec taxes, when using native magento tax rules, for a quote item the $appliedTaxes looks like something like this
```
$appliedTaxes array(2)
    ['ca-gst'] (magento\tax\model\taxdetails\appliedtax)(28)
        ['data'] array(4)
            amount = 1
            percent = 5
            tax_rate_key = "ca-gst"
            rates = array(1)
                ca-gst (magento\tax\model\taxdetails\appliedtaxrate)(28)
                    data array(3)
                        percent = 5
                        code = ca-gst
                        title = ca-gst
    ['qc-qst'] (magento\tax\model\taxdetails\appliedtax)(28)
        ['data'] array(4)
            amount = 1.99
            percent = 9.975
            tax_rate_key = "qc-qst"
            rates = array(1)
                qc-qst (magento\tax\model\taxdetails\appliedtaxrate)(28)
                    data array(3)
                        percent = 9.975
                        code = qc-qst
                        title = qc-qst
```

When using Avatax module, it seems all the taxes are added together. The same $appliedTaxes looks like this
```
$appliedTaxes array(1)
    ['ca-gst-qc-qst'] (magento\tax\model\taxdetails\appliedtax)(28)
        ['data'] array(4)
            amount = 2.99
            percent = 14.975
            tax_rate_key = "ca-gst-qc-qst"
            rates = array(2)
                ca-gst (magento\tax\model\taxdetails\appliedtaxrate)(28)
                    data array(3)
                        percent = 5
                        code = ca-gst
                        title = ca-gst
                qc-qst (magento\tax\model\taxdetails\appliedtaxrate)(28)
                    data array(3)
                        percent = 9.975
                        code = qc-qst
                        title = qc-qst
```

I don't have access to our client Avatax account so I don't know if there is another configuration to set, but with that same account, in their Magento 1 website the taxes correctly appears on two lines.

Hope this helps,

Thanks!

**Comment by erikhansen** [comment]
@marieevekroy Thanks for the additional info. It may be some time before we're able to get to this issue. In the meantime, you're welcome to submit a pull request to fix the issue.

**Comment by jeremybouvet** [comment]
Hi !

Has any update been done on this issue ? The ticket has been created about 3 months ago, and right now it makes Avatax not work properly for countries with multiple taxes (like Canada for example).

Thank you,

**Comment by rsisco** [comment]
@jeremybouvet - Can you provide details on an example where the tax calculation is incorrect?

**Comment by jeremybouvet** [comment]
Hi @rsisco ,

The calculation isn't incorrect. It's the display that is. It should be displayed on 2 lines when there is 2 taxes being applied.

The law in Canada requires that the provincial (state) and federal taxes are shown on different lines, with their amount, and not as one line.

Thank you,

**Comment by jeremybouvet** [comment]
Hi !

Any news ?

Thanks

**Comment by jeremybouvet** [comment]
Hi !

We are talking 4 months before this gets fixed and we get absolutely no support.

Can we get an ETA ?

Thanks,

**Comment by rsisco** [comment]
@jeremybouvet - I am in the final stages of developing an update that should address this need. I expect to have a new release out within the next few days.

**Comment by jeremybouvet** [comment]
Awesome news ! Thank you !!

**Comment by erikhansen** [comment]
@jeremybouvet We spent over 20 hours working on a solution for this and have finally come up with what we believe to be the best solution, considering Magento's approach for handling this summary information. 

The challenge we ran into is that when native Magento is calculating taxes for each cart item, it knows the exact amount of tax that each cart item is contributing towards the total. For example:

* product A that costs $19.99 has $1.00 of tax from "CANADA GST" and 1.99 of tax from "QUEBEC QST"
* product A that costs $19.99 has $1.00 of tax from "CANADA GST" and 1.99 of tax from "QUEBEC QST"

Magento's native tax calculation adds each of these tax breakdowns into individual "applied taxes" array items. However when we get the taxes back from Avalara's API, we only know about the breakdown for the total of the items in the cart, not for the individual cart items. Because of this, we had to do a good bit of work to get this information to display on the cart/checkout without breaking how native Magento works. I could delve into more technical details of why this was a challenge, but I don't know that it is necessary at this point.

Can you install our updated code on a stage/dev environment to see if it solves your problem? You can do that using this command (assuming you installed it via Composer to begin with):

```
composer require classyllama/module-avatax:dev-feature/AVS-421_64-add-tax-summary
```

Once you test and confirm everything is working, we can merge this into develop and include this in an official release.

Let us know if you have questions.

**Comment by jeremybouvet** [comment]
Hi !

It seems to be working properly on our end ! You can go ahead and merge.

Thanks,

**Comment by rsisco** [comment]
This issue has been resolved as of [release 1.2.2](https://github.com/classyllama/ClassyLlama_AvaTax/releases/tag/1.2.2). Closing issue.
